android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java

```diff
@@ -289,44 +299,39 @@ private void adjustTestFaucetControls(BuySendSwapUiInfo buySendSwapUiInfo) {
     @Override
     public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
         if (parent.getId() == R.id.network_spinner) {
-            String item = parent.getItemAtPosition(position).toString();
+            if (isSameSelectedNetwork(mSelectedNetwork)) return;
 
-            if (mJsonRpcService != null) {
-                mJsonRpcService.getAllNetworks(CoinType.ETH, chains -> {
-                    final NetworkInfo network = Utils.getNetworkInfoByName(item, chains);
-                    final String chainId = network.chainId;
-
-                    if (mActivityType == ActivityType.BUY) {
-                        adjustTestFaucetControls(getPerNetworkUiInfo(network));
-                    }
+            NetworkInfo networkInfo = mNetworkAdapter.getNetwork(position);
+            if (networkInfo != null) {
+                if (mActivityType == ActivityType.BUY) {
+                    adjustTestFaucetControls(getPerNetworkUiInfo(networkInfo));
+                }
 
-                    // Shall be fine regardless of activity type
-                    mFromValueText.setText("");
-                    mFromValueText.setHint("0");
+                // Shall be fine regardless of activity type
+                mFromValueText.setText("");
+                mFromValueText.setHint("0");
 
-                    mJsonRpcService.setNetwork(chainId, CoinType.ETH, (success) -> {
-                        if (!success) {
-                            Log.e(TAG, "Could not set network");
-                        }
-                        mJsonRpcService.getNetwork(CoinType.ETH, selectedNetwork -> {
-                            mCurrentSelectedNetwork = selectedNetwork;
-                            resetSwapFromToAssets();
-                        });
-                    });
-                    updateBalanceMaybeSwap(getCurrentSelectedAccountAddr());
+                mWalletModel.getCryptoModel().getNetworkModel().setNetwork(networkInfo, success -> {
+                    if (!success) {
+                        Log.e(TAG, "Could not set network");
```

Comment:
what does that log gives us? It doesn't say where does it happen, just spams user's logcat.

Reply:
plus to that we don't do any return and set `mSelectedNetwork` to some unknown value if it's not a success. Is it expected?

Reply:
I see the log was added by you https://github.com/brave/brave-core/blob/243e90089ae55626191d8f87f3312c29e155ac75/android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java#L309
if you want these kind of logs to be removed then let me know or probably update in group so all can remove these previously added logs.

Good catch, let me add a return if the current network wasn't set.

Reply:
It wasn't me who added that log, it might be I missed it on a review. Let's remove it to avoid logcat pollution as it brings nothing for troubleshooting for the end user.
That's where it has been added and not by me https://github.com/brave/brave-core/commit/e530244c61be68f532f2a4c36bda20411ebe085d